src/main/java/nextstep/ladder/domain/Line.java

```diff
@@ -2,30 +2,92 @@
 
 import nextstep.ladder.exception.LineDuplicateException;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
+import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
+import static nextstep.ladder.domain.Direction.*;
+
 public class Line {
-    private List<Boolean> points = new ArrayList<>();
+    private static final Random random = new Random();
+
+    private final List<Direction> directions;
+
+    public Line(int[] directions) {
+        this(Arrays.stream(directions)
+                .mapToObj(Direction::of)
+                .collect(Collectors.toList())
+        );
+    }
+
+    public Line(Direction direction) {
+        this(List.of(direction));
+    }
 
-    public Line(List<Boolean> points) {
-        validatePointsContinuity(points);
-        this.points = new ArrayList<>(points);
+    public Line(int countOfPerson) {
+        this(createDirections(countOfPerson));
     }
 
-    private void validatePointsContinuity(List<Boolean> points) {
-        IntStream.range(0, points.size() - 1)
-                .filter(i -> points.get(i) && points.get(i + 1))
+    public Line(List<Direction> directions) {
+        validateDirectionsContinuity(directions);
+        this.directions = new ArrayList<>(directions);
+    }
+
+    private void validateDirectionsContinuity(List<Direction> directions) {
+        IntStream.range(0, directions.size() - 1)
+                .filter(index -> isDuplicateLeftDirection(directions, index))
```

Comment:
```suggestion
                .filter(index -> directions.get(index).isLeft() && directions.get(index + 1).isLeft())
```

피드백에 대해 고민해주셔 감사합니다.
저는 이렇게 해볼것 같아요.

Reply:
```
public class Directions {
    private final List<Direction> directions;

    public Directions(List<Direction> directions) {
        this.directions = directions;
    }

    ...
}

public class Line {
    private final Directions directions;

    ...
}
```

`Line`의 책임과 `List<Direction>`의 책임을 분리해보면 어떨까요?

Reply:
분리해보았습니다!